### DISCUSSION TITLE
How to extend hono/jsx types?

### DISCUSSION THREAD

**Discussion by janat08** [honojs/hono#1767] [Q&A]
https://github.com/Desdaemon/typed-htmx/issues/1#issuecomment-1794974362

Trying to extend hono/jsx with typed-htmx. The one where I can add typed-htmx into types field isn't doing anything.

**Comment by usualoma** [comment]
Hi, @janat08!

In my environment, I did `npm install typed-htmx` and then set tsconfig.json as described in Usage and it was enabled.

https://github.com/Desdaemon/typed-htmx#usage

```json
{
    "compilerOptions": {
        "jsx": "react-jsx",
        "jsxImportSource": "hono/jsx",
        "types": [
            "typed-htmx"
        ]
    }
}
```

<img width="535" alt="image" src="https://github.com/honojs/hono/assets/30598/4da30023-7699-4e8f-bdd6-55f059f69f21">

Alternatively, if you wish to extend the interface as shown in the link, you can use a namespace “Hono” to write the following.

```ts
declare namespace Hono {
  interface HTMLAttributes {
    "hx-get"?: string;
  }
}

<html lang="en">
  <body>
    <button hx-get="/info">Get Info</button>
  </body>
</html>
```
